spec/controllers/api/events_controller_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'rails_helper'
+
+describe Api::EventsController do
+  render_views
+
+  let(:parsed_response) { JSON.parse(response.body) }
+
+  describe "index" do
+    let(:expected_event_keys) { %w(id name startDate endDate imageUrl members) }
+
+    context "when user_id param is missing" do
+      let(:missing_message) do
+        "The following param(s) are required but were not found: user_id"
+      end
+
+      it "returns error with missing param" do
+        get :index
+        expect(response.status).to eq(404)
+        expect(parsed_response.keys).to include("error")
+        expect(parsed_response.values).to include(missing_message)
+      end
+    end
+
+    context "when no pagination is passed" do
+      let!(:users) { create_list(:user, 3) }
+      let!(:user) { create(:user) }
+      let!(:events) { create_list(:event, 2) }
+
+      # Builds Event Memberships
+      before do
+        events.each do |event|
+          event.add_members(members: users.last(rand(users.length)))
+          # Ensure our main user is on all these events
+          event.add_members(members: [user])
+        end
+      end
+
+      it 'returns the expected payload' do
```

Comment:
Specs now check for stats code, keys in the responses and number of items in the response.